cmd/ncproxy/hcn_networking_test.go

```diff
@@ -340,88 +479,169 @@ func TestModifyNIC_HCN(t *testing.T) {
-		nicID             string
-		endpointName      string
 		iovPolicySettings *ncproxygrpc.IovEndpointPolicySetting
-		errorExpected     bool
+		networkCreateFunc func(string) (*hcn.HostComputeNetwork, error)
 	}
 	tests := []config{
 		{
 			name:              "ModifyNIC returns no error",
 			containerID:       containerID,
-			nicID:             testNICID,
-			endpointName:      endpointName,
+			networkCreateFunc: createTestIPv4NATNetwork,
 			iovPolicySettings: iovOffloadOn,
-			errorExpected:     false,
 		},
 		{
-			name:         "ModifyNIC returns no error when turning off iov policy",
-			containerID:  containerID,
-			nicID:        testNICID,
-			endpointName: endpointName,
+			name:              "ModifyNIC dual stack returns no error",
+			containerID:       containerID,
+			networkCreateFunc: createTestDualStackNATNetwork,
+			iovPolicySettings: iovOffloadOn,
+		},
+		{
+			name:              "ModifyNIC returns no error when turning off iov policy",
+			containerID:       containerID,
+			networkCreateFunc: createTestIPv4NATNetwork,
+			iovPolicySettings: &ncproxygrpc.IovEndpointPolicySetting{
+				IovOffloadWeight: 0,
+			},
+		},
+		{
+			name:              "ModifyNIC dual stack returns no error when turning off iov policy",
+			containerID:       containerID,
+			networkCreateFunc: createTestDualStackNATNetwork,
 			iovPolicySettings: &ncproxygrpc.IovEndpointPolicySetting{
 				IovOffloadWeight: 0,
 			},
-			errorExpected: false,
 		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(subtest *testing.T) {
+			// create test network
+			networkName := subtest.Name() + "-network"
+			network, err := test.networkCreateFunc(networkName)
+			if err != nil {
+				subtest.Fatalf("failed to create test network with %v", err)
+			}
+			defer func() {
+				_ = network.Delete()
+			}()
+
+			// create test endpoint
+			endpointName := subtest.Name() + "-endpoint"
+			endpoint, err := createTestEndpoint(endpointName, network.Id)
+			if err != nil {
+				subtest.Fatalf("failed to create test endpoint with %v", err)
+			}
+			defer func() {
+				_ = endpoint.Delete()
+			}()
+			endpointSettings := &ncproxygrpc.HcnEndpointSettings{
+				Policies: &ncproxygrpc.HcnEndpointPolicies{
+					IovPolicySettings: test.iovPolicySettings,
+				},
+			}
+			testNICID := subtest.Name() + "-nicID"
+			req := &ncproxygrpc.ModifyNICRequest{
+				ContainerID:  test.containerID,
+				NicID:        testNICID,
+				EndpointName: endpointName,
+				EndpointSettings: &ncproxygrpc.EndpointSettings{
+					Settings: &ncproxygrpc.EndpointSettings_HcnEndpoint{
+						HcnEndpoint: endpointSettings,
+					},
+				},
+			}
+
+			_, err = gService.ModifyNIC(ctx, req)
+			if err != nil {
+				subtest.Fatalf("expected ModifyNIC to return no error, instead got %v", err)
+			}
+		})
+	}
+}
+
+func TestModifyNIC_HCN_Error_InvalidArgument(t *testing.T) {
+	// support for setting IOV policy was added in 21H1
+	if osversion.Build() < osversion.V21H1 {
```

Comment:
This applies to some other tests in this file, but we don't have a syso file with the manifest info in it in this directory so I think we'd pretty much permanently skip this test as we'd spit out windows 8's build number

Reply:
if you rebase for the functional test PR, you can `_ "github.com/Microsoft/hcsshim/test/internal/manifest"`
or, itll be imported automatically if you use `"github.com/Microsoft/hcsshim/test/internal/require".Build(osversion.V21H1)`, itll manifest the exe automatically.

Reply:
We should plop a syso in here as ncproxy should probably be manifested anyways

Reply:
I used go-winres to add the syso files. @dcantah could you verify if it looks valid?